libraries/freertos_plus/aws/greengrass/test/aws_test_ggd_system.c

```diff
@@ -158,20 +197,20 @@ TEST( GGD_System, GetGGCIPandCertificate )
          * Check with auto-search flag set to false and true
          *  @{
          */
-        for( i = 0; i < ggdTestLOOP_NUMBER; i++ )
-        {
-            xStatus = GGD_GetGGCIPandCertificate( clientcredentialMQTT_BROKER_ENDPOINT,
-                                                  clientcredentialGREENGRASS_DISCOVERY_PORT,
-                                                  clientcredentialIOT_THING_NAME,
-                                                  cBuffer, /*lint !e971 can use char without signed/unsigned. */
-                                                  ulBufferSize,
-                                                  &xHostAddressData );
-
-            snprintf( cMsgBuffer, nBufferLength,
-                      "GGD_GetGGCIPandCertificate returned %d on iteration %d",
-                      ( int ) xStatus, ( int ) i );
-            TEST_ASSERT_EQUAL_INT32_MESSAGE( pdPASS, xStatus, cBuffer );
-        }
+        RETRY_EXPONENTIAL( xStatus - GGD_GetGGCIPandCertificate( clientcredentialMQTT_BROKER_ENDPOINT,
+                                                                 clientcredentialGREENGRASS_DISCOVERY_PORT,
+                                                                 clientcredentialIOT_THING_NAME,
+                                                                 cBuffer, /*lint !e971 can use char without signed/unsigned. */
+                                                                 ulBufferSize,
+                                                                 &xHostAddressData ),
+                           pdPASS,
+                           IOT_TEST_GGD_INITIAL_CONNECTION_RETRY_DELAY,
+                           IOT_TEST_GGD_CONNECT_RETRY_COUNT );
+
+        snprintf( cMsgBuffer, nBufferLength,
+                  "GGD_GetGGCIPandCertificate returned %d on iteration %d",
+                  ( int ) xStatus, ( int ) i );
```

Comment:
Doesn't seem to make sense anymore since it's not in a loop. I'm not sure what the purpose of `cMsgBuffer` was before this either since it doesn't seem to be used later. Is `i` even used anymore in this function?